### DISCUSSION TITLE
Adjusting the hotend temperature depending on the layer printing time

### DISCUSSION THREAD

**Discussion by ltwmichalw** [OrcaSlicer/OrcaSlicer#4508] [Ideas]
A few days ago, I came up with an idea for a new functionality: adjusting the printing temperature to the layer printing time or printing speed.
The idea is that when printing at low speed, the slicer can adjust the hotend temperature, for example 205 for a layer time of 10 seconds and 215 for a layer time of over 45 seconds.
This happens when printing PLA elements with different layer times. When the hotend slows down to maintain a minimum layer time, the detail becomes glossier, and when the layer is large and the hotend moves at full speed, the printed object becomes more matte. If I lower the temperature so that the entire element is matte, large layers stick to each other less well, if I raise the temperature so that the whole element is shiny, I have to increase the minimum layer time.
If it were possible to lower the temperature for short-curing layers, they could be printed faster with the same cooling.

**Comment by Rickthebig** [comment]
Cura has the "Small Layer Printing Temperature", a similar feature. It should be implemented to accomodate any part of the print needing special temperature settings.
Changing temperature during the print is a complex matter though, but developers should pay attention to the fact that at some point it will be possible to change temperature very fast. There actually is a nozzle capable of doing that, it is just too expensive right now (about 400€), can't find a link to it right now.

**Reply by ltwmichalw** [reply]
My idea is not about a quick change in temperature. Currently, I sometimes do it manually from the panel, of course when I am at the printer.
Some prints have very large differences in the size of the layers, for example a wide base and then a tower with a small diameter. In this case, changing the temperature by, for example, 5 degrees down while changing the layer size allows you to maintain a similar surface.

**Reply by JAM3SY** [reply]
You would need to account for the fact that Volumetric rate decreases with decreasing temperature. The speed attainable at a higher temp, may not be possible or cause inconsistent flow rate artifacts and lower temps.

**Reply by ltwmichalw** [reply]
Okay, now for some theory.
The thermistor measures the temperature of the hotend, not the filament that leaves the nozzle. The filement temperature is actually the resultant of the hotend temperature and the flow rate. The longer the material stays in the hotend, the more it heats up, and if it has enough time, it reaches the hotend temperature.
At high printing speeds, i.e. high flows, the filament has little time to heat up, so we raise the hotend temperature to accelerate heating.
At low speeds and low flows, the filament stays in the hotend longer, then we can reduce the hotend temperature because the filament will have more time to gain temperature.
The printing speed may be strongly limited by the minimum printing time of one layer, even up to 10 mm/s. Then high flow is unnecessary, but artifacts appear related to the change in the temperature of the dilament.

The idea is to be able to select such parameters that the filament always has a similar temperature, regardless of whether the flowrate is 5 or 30 mm3/s
